docs/design-documents/hal/0004-pin-names-general-guidelines.md

```diff
@@ -0,0 +1,132 @@
+
+    #define RED_LED = LED1
+    #define STATUS_LED = LED2 
+
+However, these names do not apply to all boards and hence should not be used in example applications that are considered platform agnostic.
+
+
+**Using LEDs at application**
+
+The detection of available LEDs at application level can be done as follow:
+
+    #ifdef(LED1)
+        DigitalOut myLED(LED1);
+        myLED = 1;
+    #endif 
+
+Alternatively, if the usage of an LED is required, then the application can detect its availability generate an error accordingly:
+
+    #ifndef(LED1)
+        #error This application requires usage of an LED
+    #endif 
+
+### Buttons
+
+**Definition of Buttons**
+
+All valid BUTTON definitions should be defined as follows in PinNames.h:
+
+    // Px_xx relates to the processor pin connected to the Button  
+    #define BUTTON1 = Px_xx  // BUTTON1  
+    #define BUTTON2 = Px_xx  // BUTTON2  
+    #define BUTTON3 = Px_xx  // BUTTON3  
+    #define BUTTON4 = Px_xx  // BUTTON4   
+    .  
+    .  
+    #define BUTTONN = Px_xx   // BUTTONN  
+
+A comment describing more BUTTON properties can also be provided. If descriptive names are required, they can be defined as aliases, as shown below:
+
+    #define PUSH_BUTTON = BUTTON1 // Momentary push Button
+
+**Using Buttons at application**
+
+The detection of available Buttons at application level can be done as follow:
+
+    #ifdef(BUTTON1)
+        DigitalIn myBUTTON(BUTTON1);
+        int input = myBUTTON.read();
+    #endif 
+
+Alternatively, if the usage of a BUTTON is required, then the application can detect its availability generate an error accordingly:
+
+    #ifndef(BUTTON1)
+        #error This application requires usage of a BUTTON
+    #endif 
+
+### Non-valid definitions
+
+If either LEDs or BUTTONs are not available, then should not be defined.
+This allows for unavailable LEDs/BUTTONs to be caught and generate an error.
+   
+    LED1 = PB_0,       // LED1 is valid
+    LED2 = LED1,       // Not valid as it's duplicate  
+    LED3 = PB_0,       // Not valid as it's duplicate 
+    LED4 = NA          // Not valid definition as LED4 does not exist
+
+    BUTTON1 = PB_1,    // BUTTON1 is valid
+    BUTTON2 = BUTTON1, // Not valid as it's duplicate  
+    BUTTON3 = PB_1,    // Not valid as it's duplicate  
+    BUTTON4 = NA,      // Not valid as BUTTON4 doesn't exist 
+
+
+### Testing compliance
+
+There should be both compile and run time checks to confirm whether a board has valid LEDs and BUTTONS defined. This can be achieved by using Greentea, for example:
+
+    mbed test -t <toolchain> -m <target> -n *test_generic_pin_names* --compile
+    mbed test -t <toolchain> -m <target> -n *test_generic_pin_names* --run
+
+If no valid pinnames for LEDs and BUTTONs are detected, then a warning should be displayed. 
```

Comment:
Clarify that's not mandatory to add LEDs and BUTTONs, but if defined, need to be compliant